.github/workflows/libc-overlay-tests.yml

```diff
@@ -0,0 +1,81 @@
+# This workflow is for pre-commit testing of the LLVM-libc project.
+name: LLVM-libc Pre-commit Overlay Tests
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - 'libc/**'
+      - '.github/workflows/libc-overlay-tests.yml'
+
+jobs:
+  build:
+    env:
+      SCCACHE_GHA_ENABLED: "true"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Set fail-fast to false to ensure that feedback is delivered for all matrix combinations.
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
I recommend using the tag names with the explicit versions (e.g. ubuntu-24.04).  That way we don't get suprised when GitHub updates the *-latest tag.

Reply:
Good point!